lib/Driver/UnixToolChains.cpp

```diff
@@ -293,23 +273,12 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   Arguments.push_back(
       context.Args.MakeArgString("--target=" + getTriple().str()));
 
-  if (getTriple().getOS() == llvm::Triple::Linux) {
-    // Make sure we only add SanitizerLibs for executables
-    if (job.getKind() == LinkKind::Executable) {
-      if (context.OI.SelectedSanitizers & SanitizerKind::Address)
-        addLinkSanitizerLibArgsForLinux(context.Args, Arguments, "asan", *this);
-
-      if (context.OI.SelectedSanitizers & SanitizerKind::Thread)
-        addLinkSanitizerLibArgsForLinux(context.Args, Arguments, "tsan", *this);
-
-      if (context.OI.SelectedSanitizers & SanitizerKind::Undefined)
-        addLinkSanitizerLibArgsForLinux(context.Args, Arguments, "ubsan", *this);
-
-      if (context.OI.SelectedSanitizers & SanitizerKind::Fuzzer)
-        addLinkRuntimeLib(context.Args, Arguments,
-                          sanitizerRuntimeLibName("fuzzer"));
-
-    }
+  // On Linux, delegate to Clang and let it figure out the correct linker
+  // options for sanitizers.
+  if (getTriple().getOS() == llvm::Triple::Linux &&
+      job.getKind() == LinkKind::Executable && context.OI.SelectedSanitizers) {
+    Arguments.push_back(context.Args.MakeArgString(
+        "-fsanitize=" + getSanitizerList(context.OI.SelectedSanitizers)));
```

Comment:
Hm. This works, but is passing through the contents of Swift's own OPT_sanitize_EQ good enough? Just in the interest of less code to maintain.

Reply:
This seems much safer, and might be worth the extra complexity.

Reply:
Okay, I can be overruled here!

Reply:
I will provide a follow-up patch to reduce the maintenance cost a bit.